src/S0A_download_ASDF_MPI.py

```diff
@@ -1,305 +1,313 @@
-    if not os.path.isfile(ff):
-        with pyasdf.ASDFDataSet(ff,mpi=False,compression="gzip-3",mode='w') as ds:
-            pass
-    else:
-        with pyasdf.ASDFDataSet(ff,mpi=False,mode='r') as rds:
-            alist = rds.waveforms.list()
-            for ista in range(nsta):
-                tname = net[ista]+'.'+sta[ista]
-                if tname in alist:
-                    num_records[ista] = len(rds.waveforms[tname].get_waveform_tags())
-
-    # appending when file exists
-    with pyasdf.ASDFDataSet(ff,mpi=False,compression="gzip-3",mode='a') as ds:
-
-        # loop through each channel
-        for ista in range(nsta):
-
-            # continue when there are alreay data for sta A at day X
-            if num_records[ista] == ncomp:
-                continue
-
-            # get inventory for specific station
-            try:
-                sta_inv = client.get_stations(network=net[ista],
-                                              station=sta[ista],
-                                              location=location[ista],
-                                              starttime=s1,
-                                              endtime=s2,
-                                              level="response")
-            except Exception as e:
-                print(e);continue
-
-            # add the inventory for all components + all time of this tation         
-            try:
-                ds.add_stationxml(sta_inv) 
-            except Exception: 
-                pass   
-
-            try:
-                # get data
-                t0=time.time()
-                tr = client.get_waveforms(network=net[ista],
-                                          station=sta[ista],
-                                          channel=chan[ista],
-                                          location=location[ista],
-                                          starttime=s1,
-                                          endtime=s2)
-                t1=time.time()
-            except Exception as e:
-                print(e,'for',sta[ista]);continue
-                
-            # preprocess to clean data  
-            print(sta[ista])
-            tr = noise_module.preprocess_raw(tr,sta_inv,prepro_para,date_info)
-            t2 = time.time()
-            tp += t2-t1
-
-            if len(tr):
-                if location[ista] == '*':
-                    tlocation = str('00')
-                else:
-                    tlocation = location[ista]
-                new_tags = '{0:s}_{1:s}'.format(chan[ista].lower(),tlocation.lower())
-                ds.add_waveforms(tr,tag=new_tags)
-
-            #if flag:
-            print(ds,new_tags);print('downloading data %6.2f s; pre-process %6.2f s' % ((t1-t0),(t2-t1)))
-
-tt1=time.time()
-print('downloading step takes %6.2f s with %6.2f for preprocess' %(tt1-tt0, tp))
-
-comm.barrier()
-if rank == 0:
-    sys.exit()
+import sys

```

Comment:
@nrich20 do you know why this entire file is showing up as different? Were there actual changes? If not you could revert to have a cleaner diff. In this state it's very hard to see if anything actually changed.

Reply:
I think the file name is having a problem (a missing `n` in `download`). This change has been made in main branch.